includes/polling/functions.inc.php

```diff
@@ -459,12 +459,18 @@ function update_application($app, $response, $metrics = [], $status = '')
     ];
 
     if ($response != '' && $response !== false) {
-        if (Str::contains($response, [
-            'Traceback (most recent call last):',
+        /**
+         * check if the response starts with an error and set that in the app_state field
+         * also set app_status to the raw response
+         */
+        if (Str::startsWith($response, [
+            'ERROR',
+            'LEGACY',
+            'UNSUPPORTED',
+            'Traceback (most recent call last):'
         ])) {
+            $data['app_status'] = $response;
             $data['app_state'] = 'ERROR';
```

Comment:
This now would result in an Error-State even if the application returns an `UNSUPPORTED` or `LEGACY`. Is this something we can live with or should that also be set correctly?

Reply:
Sorry to bring this up, but it seems as though you have bumped into some deep issues in the application code.

I just looked through the code a bit and currently it seems to confuse app_state and app_status a lot.

app_state should be a fixed set of values 'OK', 'ERROR', 'LEGACY', 'UNSUPPORTED', and 'UNKNOWN' (maybe more) to make alerting easy.

app_status seems mostly to be the output of the script, but not 100%

The update_application() function does not line up with that, confusing things.

```php
 * @param  array  $app  app from the db, including app_id
 * @param  string  $response  This should be the return state of Application polling
 * @param  array  $metrics  an array of additional metrics to store in the database for alerting
 * @param  string  $status  This is the current value for alerting
 */
function update_application($app, $response, $metrics = [], $status = '')
```

It probably should be this:

```php
 * @param  array  $app  app from the db, including app_id (Note: we should probably be using models instead of arrays)
 * @param  string  $app_status  This should be the output from the application script. Shown when an ERROR state is set.
 * @param  array  $metrics  a key value array of additional metrics to store in the database for alerting.  Must manually set _prev fields if you want them.
 * @param  string  $state The state for alerting. If not set, will attempt to detect from $response, or set to UNKNOWN
 */
function update_application($app, $app_status, $metrics = [], $state = 'UNKNOWN')
```

Reply:
Yes! That's what I was thinking... Should I rebuild that and make a sepearate PR based on this one that updates the behaviour for all other applications?